packages/material-ui-lab/src/Rating/Rating.test.js

```diff
@@ -110,4 +111,70 @@ describe('<Rating />', () => {
     checked = container.querySelector('input[name="rating-test"]:checked');
     expect(checked.value).to.equal('2');
   });
+
+  describe('<form> integration', () => {
+    before(function beforeHook() {
+      if (/jsdom/.test(window.navigator.userAgent)) {
+        // JSDOM has issues with form validation for certain elements.
+        // We could adress them individually but that doesn't add much value if we already have a working environment.
+        this.skip();
+      }
+    });
+
+    [
+      {
+        ratingProps: { name: 'rating', defaultValue: 2 },
+        formData: [['rating', '2']],
+      },
+      {
+        ratingProps: { name: 'rating', defaultValue: 2, disabled: true },
+        formData: [],
+      },
+      {
+        ratingProps: { name: 'rating', defaultValue: 2, readOnly: true },
+        // FIXME: With native <input type="radio" /> read-only values are still submitted
```

Comment:
Note: when `readOnly` is true the component doesn't render as an input, it renders as an image. It's not meant to be used in a form. Maybe the prop should have been named differently, like `display`.

Reply:
Yeah that was quite confusing for me as well when reading the implementation. Ideally we wouldn't need to do any branching based on `readOnly`. But I can see how this is easier considering how involved the markup is to make this component work.

Would like to know how this affects screen readers. It's not really an image. It's just a visual representation of a numeric value with images. Though that might not make any difference in the end.